service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCloneControlledAzureStorageContainerResourceFlightStep.java

```diff
@@ -0,0 +1,101 @@
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ResourceKeys;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+public class LaunchCloneControlledAzureStorageContainerResourceFlightStep implements Step {
+
+  private final ControlledAzureStorageContainerResource sourceResource;
+  private final String subflightId;
+  private final UUID destinationResourceId;
+  private final UUID destinationFolderId;
+
+  public LaunchCloneControlledAzureStorageContainerResourceFlightStep(
+      ControlledAzureStorageContainerResource sourceResource,
+      String subflightId,
+      UUID destinationResourceId,
+      @Nullable UUID destinationFolderId) {
+    this.sourceResource = sourceResource;
+    this.subflightId = subflightId;
+    this.destinationResourceId = destinationResourceId;
+    this.destinationFolderId = destinationFolderId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    validateRequiredEntries(
+        context.getInputParameters(),
+        JobMapKeys.AUTH_USER_INFO.getKeyName(),
+        ControlledResourceKeys.DESTINATION_WORKSPACE_ID);
+
+    var userRequest =
+        context
+            .getInputParameters()
+            .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    var destinationWorkspaceId =
+        context
+            .getInputParameters()
+            .get(ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+
+    // build input parameter map.
+    var subflightInputParameters = new FlightMap();
+    subflightInputParameters.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
+    subflightInputParameters.put(
+        ControlledResourceKeys.DESTINATION_WORKSPACE_ID, destinationWorkspaceId);
+    subflightInputParameters.put(
+        ControlledResourceKeys.CLONING_INSTRUCTIONS, sourceResource.getCloningInstructions());
+    subflightInputParameters.put(ResourceKeys.RESOURCE, sourceResource);
+    subflightInputParameters.put(
+        JobMapKeys.DESCRIPTION.getKeyName(),
+        String.format(
+            "Clone Azure Storage Container %s", sourceResource.getResourceId().toString()));
+    subflightInputParameters.put(
+        ControlledResourceKeys.DESTINATION_CONTAINER_NAME,
+        String.format(
+            "clone-%s-%s", destinationWorkspaceId, sourceResource.getStorageContainerName()));
+    subflightInputParameters.put(
+        ControlledResourceKeys.DESTINATION_RESOURCE_ID, destinationResourceId);
+    subflightInputParameters.put(ControlledResourceKeys.DESTINATION_FOLDER_ID, destinationFolderId);
+    // Do not do the policy merge on the sub-object clone
```

Comment:
Someone correct me if I'm mistaken, but as I understand it, we only want to merge policies when we are cloning a resource into an existing destination workspace (ie [using the cloneContainer API](https://workspace.dsde-dev.broadinstitute.org/swagger-ui.html#/ControlledAzureResource/cloneAzureStorageContainer)) because there is a possibility that the policies between the two workspaces could conflict. If we are cloning into a newly cloned workspace like we are in this case, then we don't need to worry about merging the policies on the storage container clone because the policies are [merged at the workspace level](https://github.com/DataBiosphere/terra-workspace-manager/blob/a64cc1723c496004454abb034525c5d2946ed984/service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneGcpWorkspaceFlight.java#L66) in the `CloneWorkspaceFlight`

Reply:
Thanks for the explanations. Can you update the comment so it includes some of this information? As it is, the comment only reiterates the line of code that follows, without explaining why.